### PR TITLE
DD-996: use column names, read xsd only once

### PR DESCRIPTION
**Pull request by lindareijnhoudt**
Fixes DD-996

# Description of changes
* Read provenance.xsd only once
* Use column names of input-csv

# How to test

# Related PRs

(Add links)

*

# Notify

@DANS-KNAW/dataversedans

### CODE REVIEW

File: src/datastation/scripts/fix_provenance_files.py [hunk 1]
```diff
@@ -22,10 +22,11 @@
...
                      'xmlns:ddm="http://easy.dans.knaw.nl/schemas/md/ddm/"'
 
+provenance_schema_doc = etree.parse('http://easy.dans.knaw.nl/schemas/bag/metadata/prov/provenance.xsd')
 
-def validate(xml_path: str, xsd_path: str) -> bool:
-    xmlschema_doc = etree.parse(xsd_path)
-    xmlschema = etree.XMLSchema(xmlschema_doc)
+
+def validate(xml_path: str) -> bool:
+    xmlschema = etree.XMLSchema(provenance_schema_doc)
```
Comment (janvanmansum) [inline src/datastation/scripts/fix_provenance_files.py:29]: Don't you want to reuse `xmlschema` across calls?

File: src/datastation/scripts/fix_provenance_files.py [hunk 2]
```diff
@@ -22,15 +22,21 @@
...
+def validate(xml_path: str) -> bool:
     try:
         xml_doc = etree.parse(xml_path)
-        result = xmlschema.validate(xml_doc)
-        return result
-    except etree.XMLSyntaxError:
+        provenance_xmlschema.assertValid(xml_doc)
+        return True
+    except etree.XMLSyntaxError as e:
+        logging.error(e)
```
Comment (janvanmansum) [inline src/datastation/scripts/fix_provenance_files.py:35]: We also expect syntax errors in many of the old provenance.xml's

File: src/datastation/scripts/fix_provenance_files.py (hunk 1 above)
Comment (lindareijnhoudt) [reply src/datastation/scripts/fix_provenance_files.py]: done

File: src/datastation/scripts/fix_provenance_files.py (hunk 2 above)
Comment (lindareijnhoudt) [reply src/datastation/scripts/fix_provenance_files.py]: done

**Pull request merged**